crates/router/src/connector/globalpay/transformers.rs

```diff
@@ -24,30 +25,190 @@ impl TryFrom<&types::PaymentsAuthorizeRouterData> for GlobalpayPaymentsRequest {
     fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
         let metadata: GlobalPayMeta =
             utils::to_connector_meta_from_secret(item.connector_meta_data.clone())?;
-        let card = item.request.get_card()?;
-        let expiry_year = card.get_card_expiry_year_2_digit();
-        Ok(Self {
-            account_name: metadata.account_name,
-            amount: Some(item.request.amount.to_string()),
-            currency: item.request.currency.to_string(),
-            reference: item.attempt_id.to_string(),
-            country: item.get_billing_country()?,
-            capture_mode: item.request.capture_method.map(|f| match f {
-                enums::CaptureMethod::Manual => requests::CaptureMode::Later,
-                _ => requests::CaptureMode::Auto,
+        let account_name = metadata.account_name;
+
+        match item.request.payment_method_data.clone() {
+            api::PaymentMethodData::Card(ccard) => Ok(Self {
+                account_name,
+                amount: Some(item.request.amount.to_string()),
+                currency: item.request.currency.to_string(),
+                reference: item.attempt_id.to_string(),
+                country: item.get_billing_country()?,
+                capture_mode: Some(requests::CaptureMode::from(item.request.capture_method)),
+                payment_method: requests::PaymentMethod {
+                    card: Some(requests::Card {
+                        number: ccard.card_number,
+                        expiry_month: ccard.card_exp_month,
+                        expiry_year: ccard.card_exp_year,
+                        cvv: ccard.card_cvc,
+                        account_type: None,
+                        authcode: None,
+                        avs_address: None,
+                        avs_postal_code: None,
+                        brand_reference: None,
+                        chip_condition: None,
+                        cvv_indicator: Default::default(),
+                        funding: None,
+                        pin_block: None,
+                        tag: None,
+                        track: None,
```

Comment:
if possible segregate the payment method specfic fields into enums. for example in cards keep the fields which are required only for card payments